### PR TITLE
feat:[SSCA-992]: Added driftSummary to OrchestrationSummaryAPI

### PR DESCRIPTION
**Pull request by shash-harness**
## Describe your changes
Added driftSummary to the orchestrationSummaryResponse, also refactored ScorecardResponse from inside Artifact Model, to a seperate scorecard summary model. 
Have added the CI changes required  for both of the above in this pr as well.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56546)

### CODE REVIEW

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger smartchecks

**Comment by shash-harness** [comment]
trigger uts

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -2116,15 +2122,65 @@ components:
...
           description: License name
         components:
           type: array
           x-stoplight:
-            id: 4tnk7avb4e2bu
+            id: jma2y8ufdmwym
           items:
             $ref: '#/components/schemas/ComponentSummary'
       description: Element of license drift
+    Drift:
```
Comment (inderschera) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:2134]: Please rename this to OrchestrationDriftSummary

File: ssca-manager/service/src/main/java/io/harness/ssca/services/OrchestrationStepServiceImpl.java [hunk 2]
```diff
@@ -169,4 +186,74 @@ private SettingsDTO getSettingsDTO(String accountId, String orgIdentifier, Strin
             SettingsDTO.Tool.builder().name(sbomProcessRequestBody.getSbomMetadata().getTool()).version("2.0").build())
         .build();
   }
+
+  private Drift getSbomDrift(String accountId, String orgId, String projectId, String orchestrationId) {
```
Comment (inderschera) [inline ssca-manager/service/src/main/java/io/harness/ssca/services/OrchestrationStepServiceImpl.java:190]: This method should be in SbomDriftService. All the necessary things needed for sbom drift should be provided by sbom drift service.

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger ut6

**Comment by shash-harness** [comment]
trigger uts

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/ssca/SscaOrchestrationStep.java [hunk 3]
```diff
@@ -54,11 +54,27 @@ protected void modifyStepStatus(Ambiance ambiance, StepStatus stepStatus, String
...
-                .maxScore(stepExecutionResponse.getArtifact().getScorecard().getMaxScore())
+      if (stepExecutionResponse.getScorecardSummary() != null) {
+        sscaArtifactMetadata.setScorecard(SscaArtifactMetadata.Scorecard.builder()
+                                              .avgScore(stepExecutionResponse.getScorecardSummary().getAvgScore())
+                                              .maxScore(stepExecutionResponse.getScorecardSummary().getMaxScore())
+                                              .build());
+      }
+
+      if (stepExecutionResponse.getDriftSummary() != null) {
+        sscaArtifactMetadata.setDrift(
```
Comment (dhruv-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/ssca/SscaOrchestrationStep.java:65]: nit: We can extract this method, such as sscaArtifactMetadata.setDrift(getDriftSummary(stepExecutionResponse.getDriftSummary()))

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/SscaArtifactMetadata.java [hunk 4]
```diff
@@ -34,11 +34,28 @@ public class SscaArtifactMetadata implements ArtifactMetadataSpec {
...
   @Value
   @Builder
   public static class Scorecard {
     String avgScore;
     String maxScore;
   }
+
+  @Value
+  @Builder
+  public static class DriftSummary {
```
Comment (dhruv-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/SscaArtifactMetadata.java:48]: Why are we defining this class twice, can it not be reused?

**Comment by shash-harness** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/ssca/Scorecard.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import io.harness.annotations.dev.HarnessTeam;
+import io.harness.annotations.dev.OwnedBy;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+@OwnedBy(HarnessTeam.SSCA)
+public class Scorecard {
```
Comment (dhruv-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/ssca/Scorecard.java:19]: Why are we defining the same dto twice? Can we reuse the dto?

Comment (shash-harness) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/ssca/Scorecard.java:19]: Had discussed this with Reetika, she told to create a seperate one, for ArtifactEntity in beans.

**Review by inderschera — APPROVED** [review]

**Comment by shash-harness** [comment]
trigger uts

**Comment by shash-harness** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/SscaArtifactMetadata.java (hunk 4 above)
Comment (shash-harness) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/SscaArtifactMetadata.java]: The beans Scorecard DTO is on the server side, and the others are on the client side, to keep them seperate have defined in different packages.

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/ssca/Scorecard.java (hunk 5 above)
Comment (shash-harness) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/stepstatus/artifact/ssca/Scorecard.java:19]: The beans Scorecard DTO is on the server side, and the others are on the client side, to keep them seperate have defined in different packages.

**Review by dhruv-harness — APPROVED** [review]

**Comment by shash-harness** [comment]
trigger checkstylecodeformat

**Comment by shash-harness** [comment]
trigger uts

**Review by harshjain12 — APPROVED** [review]

**Comment by shash-harness** [comment]
trigger uts

**Review by inderschera — APPROVED** [review]

**Pull request merged**